### DISCUSSION TITLE
What is the best method for creating an animation?

### DISCUSSION THREAD

**Discussion by NerdyCrafter** [Bodmer/TFT_eSPI#2133] [Q&A - Sprites] [answered]
I want to make an animation with multiple images that is a little complex. I'm going to need to layer the images in different ways while moving/animating them. I was wondering though what's the best method for this. Should I just make individual sprites for each image I use and then just push them to the background sprite. Or would it be better to make a sprite that acts as a layer, taking up the full screen, and I push multiple images to that sprite and then push it to the background as transparent? I hope that makes since.

I know there are a multiple ways it could be done, just wondering if one way is best for this. I appreciate everyone help! Thanks so much Bodmer for this fantastic library!

**Comment by Bodmer** [comment, marked as answer]
The best way is one that works!
Individual sprites would be best to save memory.
Full screen buffer might be best if you have lots  and lots of RAM and the full screen update speed is adequate.
Creating an animated GIF image is another option, there is a [library here](https://github.com/bitbank2/AnimatedGIF) for that that includes a  TFT_eSPI compatible example.

**Reply by NerdyCrafter** [reply]
Thank you! I just wanted to be sure so I didn't finish programming just to find that there was a much easier method. Your work has helped me a lot, thank you so much for making this amazing library!
